ouroboros-network/test/Test/Mux.hs

```diff
@@ -390,61 +516,269 @@ prop_mux_starvation response0 response1 =
                  a <- readTBQueue q
                  flushTBQueue q (a : acc)
 
--- | Send message for an unknown miniprotocol and verify that the correct exception is thrown.
-prop_mux_unknown_miniprot :: InvalidSDU
-                          -> Property
-prop_mux_unknown_miniprot badSdu = ioProperty $ do
-    (client_w, serverAsync) <- failingServer
-
-    atomically $ writeTBQueue client_w (encodeInvalidMuxSDU badSdu)
-
-    res <- wait serverAsync
-    case res of
-         Just e  ->
-             case fromException e of
-                  Just me -> return $ Mx.errorType me == Mx.MuxUnknownMiniProtocol
-                  Nothing -> return False
-         Nothing -> return False
-
 encodeInvalidMuxSDU :: InvalidSDU -> BL.ByteString
-encodeInvalidMuxSDU sdu = Bin.runPut enc
+encodeInvalidMuxSDU sdu =
+    let header = Bin.runPut enc in
+    BL.append header $ BL.replicate (fromIntegral $ isLength sdu) 0xa
```

Comment:
Is there something special about `0xa` here, or could we generate an arbitrary `ByteString`?

Reply:
There is nothing special about `0xa`.
Using your approach to generate an arbitrary would be possible but again, it is too expensive.
I will change `InvalidSDU` to take an arbitrary Word8 pattern.